src/Chainweb/Miner/Pact.hs

```diff
@@ -159,8 +162,14 @@ readRewards =
         <> sshow e
       Right vs -> MinerRewards $ M.fromList . V.toList . V.map formatRow $ vs
   where
-    formatRow :: (Word64, Double) -> (BlockHeight, Decimal)
-    formatRow (!a,!b) = (BlockHeight $ int a,  fromRational $ toRational b)
+    formatRow :: (Word64, CsvDecimal) -> (BlockHeight, Decimal)
+    formatRow (!a,!b) = (BlockHeight $ int a, _csvDecimal b)
+
+newtype CsvDecimal = CsvDecimal { _csvDecimal :: Decimal }
+    deriving newtype (Show, Read)
+
+instance CSV.FromField CsvDecimal where
+    parseField = either fail pure . readEither . B8.unpack
```

Comment:
In my experience it's usually a bad idea to use `Data.ByteString.Char8` due to incorrect edge cases with Unicode.

Reply:
I think, it should be fine here. The input is ASCII and `readEither` should fail if it isn't.

We could interpret the input as `Text` and do an Utf8 encoding roundtrip to filter out any nasty code points or deal with non-ASCII decimal digit representation, but that seems overkill to me here.